File: Hello World in Several Languages/Hello.java
public class Hello {
    public static void main(String[] args) {
        System.out.println("Hello Java!");
        Hello greeter = new Hello();
        for(int i = 0; i < args.length; ++i)
            greeter.greet(args[i]);
    }

    public void greet(String name) {
        System.out.println("Hello " + name + "!");
    }
}

File: Hello World in Several Languages/hello.r
hello <- function(name) {
    cat("Hello", name, "\n")
}

main <- function() {
    cat("Hello R!\n")
    args <- commandArgs(trailingOnly = TRUE)
    for(i in 1:length(args)) {
        hello(args[i])
    }
}

main();
